src/day4.scala

```diff
@@ -0,0 +1,55 @@
+// using scala 3.0.2
+
+package day4
+
+import scala.io.Source
+
+@main def run(): Unit =
+  val input = util.Using.resource(Source.fromFile("input/day4"))(_.mkString)
+  val (part1, part2) = answers(input)
+  println(s"The answer of part1 is $part1")
+  println(s"The answer of part2 is $part2")
+
+case class Board(lines: List[List[Int]]):
+  def mapNumbers(f: Int => Int): Board = Board(lines.map(_.map(f)))
+  def columns: List[List[Int]] = lines.transpose
+
+object Board:
+  def parse(inputBoard: String): Board =
+    val lines = inputBoard.split('\n').toList
+    Board(lines.map(parseLine))
+
+  private val lineParser = raw"\d+".r
+  private def parseLine(inputLine: String): List[Int] =
+    lineParser.findAllIn(inputLine).toList.map(_.toInt)
+
+def answers(input: String): (Int, Int) =
+  val inputSections: List[String] = input.split("\n\n").toList
+  val numbers = inputSections.head.split(',').map(_.toInt)
+
+  val numberToTurn = numbers.zipWithIndex.toMap
```

Comment:
This only works if numbers are never drawn twice. This is not guaranteed by the problem statement, so we should not rely on it. We must only keep the first turn at which any given number is drawn.

Reply:
Indeed it is not stated in the puzzle description but we assume it is implicit. Indeed it seems that everyone's input contains all numbers from `0` to `99`. Even in the example the input contains all numbers from `0` to `26`. @sporarum will state it clearly in its blog post.